include/persistence.h

```diff
@@ -499,6 +508,24 @@ template <typename Complex> class persistence_computer_t {
 		}
 	}
 
+#ifdef RETRIEVE_PERSISTENCE
+	index_t get_euler_characteristic() { return euler_characteristic; }
+
+	std::vector<size_t> get_betti_numbers() { return betti_numbers; }
+
+	size_t get_betti_numbers(size_t dimension) { return betti_numbers[dimension]; }
+
+	std::vector<std::vector<std::pair<value_t, value_t>>> get_persistence_diagram() { return birth_deaths_by_dim; }
```

Comment:
How about returning a const reference instead to avoid copying? Users can then copy themselves if they want to.
```suggestion
	const std::vector<std::vector<std::pair<value_t, value_t>>>& get_persistence_diagram() const { return birth_deaths_by_dim; }
```